public/js/actions/transaction.js

```diff
@@ -27,9 +27,9 @@ export function payWithNewCard() {
 export function _processOneTimePayment({dispatch, productId, getState,
                                         payNonce, payMethodUri,
                                         fetch=api.fetch,
-                                        amount}) {
+                                        userDefinedAmount}) {
   var data = {
-    amount: amount,
+    amount: userDefinedAmount,
```

Comment:
while you're in here, you might as well wrap this in an if statement + log statement like you did with the subscription. This will prepare us for processing one-time payments that aren't for donations.